recipes/libmysqlclient/all/conanfile.py

```diff
@@ -20,36 +19,83 @@ class LibMysqlClientCConan(ConanFile):
 
     _cmake = None
 
+    @property
+    def _with_zstd(self):
+        return tools.Version(self.version) > "8.0.17"
+
+    @property
+    def _with_lz4(self):
+        return tools.Version(self.version) > "8.0.17"
+
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
     def requirements(self):
         if self.options.with_ssl:
-            self.requires("openssl/1.1.1j")
+            self.requires("openssl/1.1.1k")
 
         if self.options.with_zlib:
             self.requires("zlib/1.2.11")
+        if self._with_zstd:
+            self.requires("zstd/1.5.0")
+        if self._with_lz4:
+            self.requires("lz4/1.9.3")
 
     def source(self):
-        archive_name = "mysql-" + self.version
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename(archive_name, self._source_subfolder)
-        for lib in ["icu", "libevent", "re2", "rapidjson", "lz4", "protobuf", "libedit"]:
-            tools.rmdir(os.path.join(self._source_subfolder, "extra", lib))
-        for folder in ['client', 'man', 'mysql-test']:
-            tools.rmdir(os.path.join(self._source_subfolder, folder))
-        tools.rmdir(os.path.join(self._source_subfolder, "storage", "ndb"))
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
 
     def _patch_files(self):
-        for patch in self.conan_data["patches"][self.version]:
+        libs_to_remove = ["icu", "libevent", "re2", "rapidjson", "protobuf", "libedit"]
```

Comment:
protobuf?

Reply:
yes ? it was already removed in https://github.com/conan-io/conan-center-index/pull/5677/files#diff-e41df3cd040880d0fbfd4ee3f7d9da30057633dbf02edd67075926c558ff17c6L17, I just made this generic for all verions of libmysqlclient

Reply:
Ahhh I missed that... SQL and protobuf.... sounds complicated!

Thanks for clarifying